packages/render-fidelity-tools/src/workflows/update-screenshots.ts

```diff
@@ -39,12 +39,28 @@ const goldensDirectory = join(rootDirectory, 'goldens');
 const renderersDirectory = join(rootDirectory, 'renderers')
 
 let scenarioWhitelist: Set<string>|null = null;
+let rendererWhitelist: Set<string>|null = null;
+const rendererList = new Set(config.renderers.map((renderer: any) => {
+  return renderer.name;
+}));
 
+// default update screenshots command takes 4 arguments. If there's more than 4,
+// user has specify either scenarios or renderers
 if (process.argv.length > 3) {
-  scenarioWhitelist = new Set();
-
   for (let i = 3; i < process.argv.length; i++) {
-    scenarioWhitelist.add(process.argv[i]);
+    const argName = process.argv[i];
+
+    if (rendererList.has(argName)) {
+      if (rendererWhitelist === null) {
+        rendererWhitelist = new Set();
+      }
+      rendererWhitelist.add(argName);
+    } else {
+      if (scenarioWhitelist === null) {
+        scenarioWhitelist = new Set();
+      }
+      scenarioWhitelist.add(argName);
+    }
```

Comment:
Do you think i use too many nested if here?

Reply:
Looks fine to me.